lib/Stats.js

```diff
@@ -384,11 +430,14 @@ class Stats {
 						}
 						if (chunk.name) {
 							assetsByFile[asset].chunkNames.push(chunk.name);
-							if (obj.assetsByChunkName[chunk.name])
-								obj.assetsByChunkName[chunk.name] = []
-									.concat(obj.assetsByChunkName[chunk.name])
-									.concat([asset]);
-							else obj.assetsByChunkName[chunk.name] = asset;
+							if (obj.assetsByChunkName[chunk.name]) {
+								obj.assetsByChunkName[chunk.name] = [
```

Comment:
I had to change this because of:

```
error TS2345: Argument of type 'any[]' is not assignable to parameter of type 'ConcatArray<never>'.
```

Reply:
This is because you initiated the array with empty array literal and no types. You can resolve it by defining types

Reply:
I wonder why an empty array literal is considered as `Array<never>` instead of `Array<any>` here.

Even when this is correct, why doesn't `concat` add the argument type. like this:

``` typescript
Array<T>.concat(arg: Array<U>) : Array<T | U>
```

Reply:
Anyway, the syntax change is fine.

Reply:
@sokra 
1. The compiler tries not to introduce `any`, especially with strict null checks, and `Array<never>` is *technically* correct, although not really useful. But the syntax change in this PR is a much better solution in my opinion.
2. We assume that people want homogeneous arrays when they're using types, since putting the wrong kind of thing in an array is a common bug.